File: LeetCode/1436. Destination City/Solution.php
class Solution {
  function destCity($paths) {
      $destinations = array();
  
      foreach ($paths as $p) {
          $destinations[$p[0]] = $p[1];
      }
  
      $currentCity = $paths[0][1];
  
      while (array_key_exists($currentCity, $destinations)) {
          $currentCity = $destinations[$currentCity];
      }
  
      return $currentCity;
  }
}

File: LeetCode/1436. Destination City/Solution.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_map<string, string> destinations;
        
        for (int i = 0; i < paths.size(); i++) {
            destinations.insert({paths[i][0], paths[i][1]});
        }

        string currentCity = paths[0][0];

        while (destinations.find(currentCity) != destinations.end()) {
            currentCity = destinations[currentCity];
        };
        
        return currentCity;
    }
};
